realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -66,17 +66,18 @@ JNIEXPORT jstring JNICALL Java_io_realm_internal_TableQuery_nativeValidateQuery(
 // If the corresponding entry in tablesArray is anything other than a nullptr, the link is a backlink.
 // In that case, the tablesArray element is the pointer to the backlink source table and the
 // indicesArray entry is the source column index in the source table.
+// FIXME!!!  This doesn't actually seem to be following backlinks.
```

Comment:
I think it's `getTableByArray()` which doesn't follow backlinks. See https://github.com/realm/realm-core/pull/2597